test/test_opsum_to_ttn.jl

```diff
@@ -125,19 +141,78 @@ using Test
     end
   end
 
+  @testset "OpSum to TTN Fermions" begin
+    # small comb tree
+    auto_fermion_enabled = ITensors.using_auto_fermion()
+    if !auto_fermion_enabled
+      ITensors.enable_auto_fermion()
+    end
+    tooth_lengths = fill(2, 3)
+    c = named_comb_tree(tooth_lengths)
+    is = siteinds("Fermion", c; conserve_nf=true)
+
+    # test with next-nearest neighbor tight-binding model
+    t = 1.0
+    tp = 0.4
+    U = 0.0
+    h = 0.5
+    H = ITensorNetworks.tight_binding(c; t=t, tp=tp, h=h)
+
+    # add combination of longer range interactions
+    Hlr = copy(H)
+
+    @testset "Svd approach" for root_vertex in leaf_vertices(is)
+      # get TTN Hamiltonian directly
+      Hsvd = TTN(H, is; root_vertex=root_vertex, cutoff=1e-10)
+      # get corresponding MPO Hamiltonian
+      sites = only.([is[v] for v in reverse(post_order_dfs_vertices(c, root_vertex))])
+      vmap = Dictionary(reverse(post_order_dfs_vertices(c, root_vertex)), 1:length(sites))
+      Hline = ITensors.MPO(relabel_sites(H, vmap), sites)
+      # compare resulting sparse Hamiltonians
+      Hmat_sp = hopping_hamiltonian(relabel_sites(H, vmap))
+      @disable_warn_order begin
+        Tmpo = prod(Hline)
+        Tttno = contract(Hsvd)
+      end
+
+      # verify that the norm isn't 0 and thus the same (which would indicate a problem with the autofermion system
+      @test norm(Tmpo) > 0
+      @test norm(Tttno) > 0
+      @test norm(Tmpo) ≈ norm(Tttno) rtol = 1e-6
+
+      @test_broken Tmpo ≈ Tttno # ToDo fix comparison for fermionic tensors
+      # In the meantime: matricize tensors and convert to dense Matrix to compare element by element
+      Tmm = _fermionic_to_dense_matrix(Tmpo)
+      Ttm = _fermionic_to_dense_matrix(Tttno)
+      Tmm, Ttm = _make_inds_same(Tmm, Ttm)
+
+      dTmm = Matrix(dense(Tmm), inds(Tmm)[1], inds(Tmm)[2])
+      dTtm = Matrix(dense(Ttm), inds(Tmm)[1], inds(Tmm)[2])
```

Comment:
Code like this and `_make_inds_same` that relies on indices being ordered in a certain way generally isn't a good idea.

It seems like what you want is a function that takes an ITensor that has pairs of primed and unprimed indices and turns it into a matrix from primed to unprimed indices.

So maybe:
```julia
function to_matrix(t::ITensor)
  c = combiner(inds(t; plev=0))
  tc = t * c * dag(c')
  @assert order(tc) == 2
  cind = combinedind(c)
  return matrix(tc, cind', cind)
end
```
and then
```julia
  dTmm = to_matrix(Tmpo)
  dTtm = to_matrix(Tttno)
```

Reply:
Ok. I made a specialized function to the same thing to two tensors and replace the indices of one of them, which doesn't look as nice as what you suggested (but addresses the stability issue).

Reply:
What do you mean by "stability issue"? I prefer the simplicity of the version I proposed.